sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/QueryResult.cs

```diff
@@ -62,13 +55,21 @@ internal static QueryResult<T> DeserializeQueryResult(JsonElement element, Objec
                     {
                         continue;
                     }
+
                     var array = new List<T>();
+
                     foreach (JsonElement item in property.Value.EnumerateArray())
                     {
-                        using MemoryStream streamedObject = StreamHelper.WriteToStream(item, objectSerializer, default);
+                        // defaultObjectSerializer of type JsonObjectSerializer needs to be used to serialize the JsonElement into a stream.
+                        // Using any other ObjectSerializer (e.g. NewtonsoftJsonObjectSerializer) won't be able to deserialize the JsonElement into
+                        // a MemoryStream correctly.
+                        using MemoryStream streamedObject = StreamHelper.WriteToStream(item, defaultObjectSerializer, default);
```

Comment:
@pakrym if we could use a JsonElement or a string instead of a stream to feed into the object serializer, we wouldn't have to go through these hoops. Any thoughts?

Reply:
We would still have to turn the JsonElement into the string/stream before feeding to Newtonsoft.Json Parse.

We are adding a BinaryData overload in this release though.